### PR TITLE
Update func isRelativeLink

### PR DESCRIPTION
**Pull request by elian0211**
when link to current directory or parent directory

### CODE REVIEW

File: inline_test.go [hunk 1]
```diff
@@ -72,7 +72,12 @@ func doTestsInlineParam(t *testing.T, tests []string, extensions, htmlFlags int,
 		input := tests[i]
 		candidate = input
 		expected := tests[i+1]
+		expected = strings.Replace(expected, "\r\n", "\n", -1)
+		expected = strings.Replace(expected, "\r", "\n", -1)
```
Comment (shurcooL) [inline inline_test.go]: Why is this added?

Comment (elian0211) [inline inline_test.go]: A newline on Windows equals "\r\n"
on Mac OS  : "\r"

Comment (russross) [inline inline_test.go]: Line ending conventions are not relevant here, since the test cases are
fully specified in the repository files. Nothing is being generated by an
external tool where line endings might vary by host OS.

Comment (shurcooL) [inline inline_test.go]: Yeah, that's why I was puzzled seeing this added in this PR.

File: upskirtref_test.go [hunk 2]
```diff
@@ -49,9 +50,14 @@ func doTestsReference(t *testing.T, files []string, flag int) {
 			continue
 		}
 		expected := string(expectedBytes)
+		expected = strings.Replace(expected, "\r\n", "\n", -1)
+		expected = strings.Replace(expected, "\r", "\n", -1)
 
 		// fmt.Fprintf(os.Stderr, "processing %s ...", filename)
 		actual := string(runMarkdownReference(input, flag))
+		actual = strings.Replace(actual, "\r\n", "\n", -1)
+		actual = strings.Replace(actual, "\r", "\n", -1)
```
Comment (shurcooL) [inline upskirtref_test.go]: 0d9aa1ca7ef663f9e67a7031a31370f52e2ac9a6 removed one instance, but this other one remains.

**Comment by rtfb** [comment]
Thanks @elian0211. Could you please resubmit the PR as one or two well-formed commits on a clean branch? The current state is way too messy.

**Comment by elian0211** [comment]
OK
